frost-coordinator/src/coordinator.rs

```diff
@@ -62,19 +97,125 @@ where
         result
     }
 
-    pub fn sign_message(&mut self, _msg: &str) -> Result<(), Error> {
+    pub fn sign_message(&mut self, msg: &[u8]) -> Result<(), Error> {
+        if self.aggregate_public_key == Point::default() {
+            return Err(Error::NoAggregatePublicKey);
+        }
+
         let nonce_request_message = Message {
-            msg: MessageTypes::NonceRequest(NonceRequest { dkg_id: 0 }),
+            msg: MessageTypes::NonceRequest(NonceRequest {
+                dkg_id: self.current_dkg_id,
+            }),
             sig: [0; 32],
         };
 
         self.network.send_message(nonce_request_message)?;
 
-        todo!();
+        loop {
+            match self.wait_for_next_message()?.msg {
+                MessageTypes::NonceRequest(_) => {}
+                MessageTypes::NonceResponse(nonce_response) => {
+                    self.public_nonces
+                        .insert(nonce_response.signer_id as u32, nonce_response);
+                }
+                msg => {
+                    println!("Got unexpected message {:?})", msg);
+                }
+            }
+
+            if self.public_nonces.len() == self.threshold {
+                break;
+            }
+
+            println!("Got {} nonce responses", self.public_nonces.len());
```

Comment:
Nit: Use tracing logging facilities
```suggestion
            tracing::info!("Got {} nonce responses", self.public_nonces.len());
```